pkg/segment/writer/packer.go

```diff
@@ -672,6 +680,38 @@ func encJsonNumber(key string, numType SS_IntUintFloatTypes, intVal int64, uintV
 	return valSize
 }
 
+func (ss *SegStore) updateColValueSizeInAllSeenColumns(colName string, size uint32) {
+	currentSize, ok := ss.AllSeenColumns[colName]
+	if !ok {
+		if ss.RecordCount > 0 {
+			// Not the first record in the segment, so the col value size is not consistent.
+			ss.AllSeenColumns[colName] = -1
+			currentSize = -1
```

Comment:
we don't need to set currentSize and just return from here. (note see below comment, we can just have 1 return statement for this whole if !ok block